e2e/tests/test_e2e.py

```diff
@@ -1018,13 +1068,29 @@ def exec_with_kubectl(self, pod, cmd):
                               stdout=subprocess.PIPE,
                               stderr=subprocess.PIPE)
 
+    def get_patroni_state(self, pod):
+        return json.loads(self.exec_with_kubectl(pod, "patronictl list -f json").stdout)
+
+    def get_patroni_running_members(self, pod):
+        result = self.get_patroni_state(pod)
+        return list(filter(lambda x: x["State"]=="running", result))
+    
+    def get_statefulset_image(self, label_selector="application=spilo,cluster-name=acid-minimal-cluster", namespace='default'):
+        ssets = self.api.apps_v1.list_namespaced_stateful_set(namespace, label_selector=label_selector, limit=1)
+        if len(ssets.items) == 0:
+            return None
+        return ssets.items[0].spec.template.spec.containers[0].image
+
     def get_effective_pod_image(self, pod_name, namespace='default'):
         '''
         Get the Spilo image pod currently uses. In case of lazy rolling updates
         it may differ from the one specified in the stateful set.
         '''
         pod = self.api.core_v1.list_namespaced_pod(
             namespace, label_selector="statefulset.kubernetes.io/pod-name=" + pod_name)
+        
+        if len(pod.items) == 0:
+            return None
```

Comment:
It is better to throw an exception in this case, That would enable tests to fail fast.

Reply:
Since this is anyways wrapped into `eventuallyEqual` that does not make a big different and would potentially if ever make handling no pod a bit more cumbersome.